x/gov/types/v1beta2/msgs.go

```diff
@@ -68,11 +60,10 @@ func (m MsgSubmitProposal) ValidateBasic() error {
 		return sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, deposit.String())
 	}
 
-	// Empty messages are not allowed
-	// TODO: ValidateBasic should check that either metadata or length is non nil
-	// if m.Messages == nil || len(m.Messages) == 0 {
-	// 	return ErrNoProposalMsgs
-	// }
+	// Check that either metadata or Msgs length is non nil.
+	if len(m.Messages) == 0 && len(m.Metadata) == 0 {
+		return sdkerrors.Wrap(types.ErrNoProposalMsgs, "either metadata or Msgs length must be non-nil")
```

Comment:
Want to double-check if we're okay with this logic

Reply:
This was what was agreed at the governance call a few weeks back